src/pages/api/webhooks/image-resource-training.ts

```diff
@@ -63,28 +64,33 @@ export default WebhookEndpoint(async (req, res) => {
 
   const status = {
     Success: TrainingStatus.InReview,
-    Fail: TrainingStatus.Failed,
     Update: TrainingStatus.Processing,
+    Fail: TrainingStatus.Failed,
+    Reject: TrainingStatus.Failed,
+    LateReject: TrainingStatus.Failed,
   }[data.type];
 
   switch (data.type) {
     case 'Success':
     case 'Fail':
+    case 'Reject':
+    case 'LateReject':
     case 'Update':
-      // TODO: this validation could be done by zod but would need to handle all the other type cases
       if (!data.context) {
         return res.status(400).json({ ok: false, error: 'context is undefined' });
       }
 
-      if (!data.context?.modelFileId) {
-        return res.status(400).json({ ok: false, error: 'modelFileId is undefined' });
+      try {
+        await updateRecords(data.context, status as TrainingStatus);
+      } catch (e: unknown) {
+        return res.status(500).json({ ok: false, error: (e as Error)?.message });
       }
 
-      await updateRecords(data.context, status as TrainingStatus);
-
       break;
+    // cannot do anything with Expire and Claim until we store the jobId
+    case 'Expire':
+    case 'Claim':
     default:
```

Comment:
We could return a 400 here stating that the `data.type` is not supported since we allow any string
```suggestion
    default:
      return res.status(400).json({ ok: false, error: 'type not supported' });
```

Reply:
thanks @manuelurenah - went ahead and did that.